packages/react/src/components/NoticeChoice/NoticeChoice.js

```diff
@@ -0,0 +1,497 @@
+  ]);
+  /**
+   * The function is called when the user changes the country. It checks if the country is English,
+   * and if so, it sets the pre-text to the default pre-text
+   */
+  const countryChangeAction = useCallback(() => {
+    const splitValue = locale.split('-', 2);
+    const ncData = getNcContentFromWindow();
+    if (splitValue[1] === 'en') {
+      let preText = defaultPreText.current;
+      if (ncData.OtherPreferences.englishNoticeText !== '') {
+        preText = ncData.OtherPreferences.englishNoticeText;
+      }
+      setPreText(preText);
+    }
+    /**
+     * @description if the user already interacted with the checkboxes,
+     * skip country default selection.
+     */
+    if (!changed) {
+      /**
+       * @description
+       * change checkbox checked option based on new country.
+       */
+      // console.log('Calling setDefaultSelections from countryChangeAction');
+      setDefaultSelections();
+    }
+  }, [locale, changed, setDefaultSelections]);
+  const checkBoxChange = (checked, id) => {
+    // event.preventDefault();
+    const newValues = {
+      ...values,
+    };
+    newValues[id] = !!checked;
+    setValues(newValues);
+    setChanged(true);
+    setPrefChange(true);
+    const fieldName = `NC_CHECK_${id}`;
+    const hiddenFieldName = `NC_HIDDEN_${id}`;
+    onChange(fieldName, checked);
+    onChange(hiddenFieldName, checked ? 'PERMISSION' : 'SUPPRESSION');
+  };
+  let customPostText = postText;
+  const ncHeading = useCallback(() => {
+    return (
+      <p
+        id="ncHeading"
+        className={`${prefix}--nc__pre-text`}
+        dangerouslySetInnerHTML={{ __html: preText }}
+      />
+    );
+  }, [preText]);
+  return featureFlag(
+    DDS_NOTICE_CHOICE,
+    <section
+      data-autoid={`${stablePrefix}--nc`}
+      className={`${prefix}--nc ${classNames}`}>
+      {loaded ? (
+        <div>
+          {ncHeading()}
+          <div className={`${prefix}--checkbox-group`}>
+            {checkboxes &&
+              Object.keys(checkboxes).length > 0 &&
+              Object.keys(checkboxes).map(key => {
+                const checked = values[key];
+                return (
+                  <span key={key}>
+                    <Checkbox
+                      {...checkboxes[key]}
+                      checked={checked}
+                      onChange={checkBoxChange}
+                      key={key}
+                      className={`${prefix}--nc__checkbox-${key}`}
+                    />
+                  </span>
+                );
+              })}
+          </div>
+          <div
+            className={`${prefix}--nc__post-text`}
+            dangerouslySetInnerHTML={{ __html: customPostText }}
+          />
+        </div>
+      ) : (
+        <>
+          <SkeletonText />
+          <SkeletonText />
+          <SkeletonText />
+          <SkeletonText />
+        </>
+      )}
+      {bpidLegalText && <BPIDLegalText bpidLegalText={bpidLegalText} />}
```

Comment:
```suggestion
      <p>{bpidLegalText}</p>;
```

would this be sufficient or is the `<BPIDLegalText />` component necessary?

Reply:
In the future, there might be custom logic inside BPIDLegalText component.